package/tests/test_integrations/test_sqlite_store.py

```diff
@@ -67,9 +216,138 @@ def test_update_git_branch(self, store_path, mocker):
         }
         mocker.patch("git.Repo.active_branch").name = "test_branch"
 
-        assert sqlite_store.to_json() == json.dumps(
+        assert sqlite_store._to_json() == json.dumps(
             {
                 "project_path": str(store_path),
                 "git": {"commit_sha": "123456", "branch": "test_branch"},
             }
         )
+
+    def test_upload_to_s3_success(self, mocker, store_path, remote_path):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._upload()
+        sqlite_store._remote_fs.put.assert_called_once()
+
+    def test_upload_to_s3_fail(self, mocker, store_path, remote_path):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.put.side_effect = ConnectionError("Connection error")
+        mock_log = mocker.patch.object(logging.Logger, "exception")
+        sqlite_store._upload()
+        mock_log.assert_called_once_with("Upload failed: Connection error")
+
+    def test_download_from_s3_success(
+        self, mocker, store_path, remote_path, mocked_db_in_s3, tmp_path
+    ):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.glob.return_value = mocked_db_in_s3
+        sqlite_store._download()
+        downloaded_dbs = set(Path(sqlite_store.location).parent.glob("*.db")) - {
+            Path(sqlite_store.location)
+        }
+        assert downloaded_dbs == {Path(tmp_path / "db1.db"), Path(tmp_path / "db2.db")}
+
+    def test_download_from_s3_failure(self, mocker, store_path, remote_path):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.glob.side_effect = ConnectionError("Connection error")
+        mock_log = mocker.patch.object(logging.Logger, "exception")
+        sqlite_store._download()
+        downloaded_dbs = set(Path(sqlite_store.location).parent.glob("*.db")) - {
+            Path(sqlite_store.location)
+        }
+        # Assert that the number of databases downloaded is 0
+        assert len(downloaded_dbs) == 0
```

Comment:
Similar comment here to above test about a better `assert` that doesn't copy and paste `downloaded_dbs` from the implementation of SQLitestore.